File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        vector<int>ans(nums.size() - k + 1,-1);
        int n = nums.size();
        
        for(int i=0;i<nums.size();i++){
            
            int j = i+1;
           
            
            int maxi = 0;
            int a = 1;
            maxi = max(maxi,nums[i]);
            while(j<nums.size() && a<k &&  nums[j] - nums[j-1]==1) {
                maxi = max(maxi ,nums[j]);
                a++;
                j++;
                
            }
            if(a==k)
            ans[i] = maxi;
            
        }
        return ans;
    }
};

File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.java
class Solution {
    public int[] resultsArray(int[] nums, int k) {
        if(k==1) return nums;
        int [] ans= new int[nums.length-k+1];
        if(nums.length==1){
            ans[0]=nums[0];
            return ans;
        }
        Arrays.fill(ans,-1);
        int i=0;
        int j=1;
        while(j<nums.length){
            while(j<nums.length && nums[j]==nums[j-1]+1){
                if(j-i+1>=k){
                    ans[j-k+1]=nums[j];
                }
                j++;
            }
            i=j;
            j=i+1;
        }
        return ans;
    }
}